src/fallback.cpp

```diff
@@ -239,6 +239,8 @@ int fish_get_emoji_width(wchar_t c) {
 #include "widecharwidth/widechar_width.h"
 
 int fish_wcwidth(wchar_t wc) {
+    wc = character_to_render(wc);
```

Comment:
Why is this here?

Wouldn't it check a different character from what would then be inserted (if it doesn't hit the "return c" case)?

Reply:
You're right, this breaks things :cold_sweat: 

Without this line, fish would assume width `0` for the tab character, which leads to drawing a command line like

![image](https://user-images.githubusercontent.com/6853872/91635737-2d0b4d80-e9fb-11ea-98c0-4fafffb79bfd.png)

We have some places where we compute the width of a tab, for example if the prompt contains a tab.
In that case, `fish_wcwidth` should not be called for the tab character.

However, for other control characters we don't have such logic so this computes the wrong width, as can be seen by typing a multi-line commandline with this prompt:

```fish
function fish_prompt
    printf (string repeat -n 50 \b)'> '
end
```

We might want to let the caller decide whether the character is something to be rendered on the command line, or rendered directly. Not sure if that will cause any trouble.

The most important thing to fix here is \t, so only pretty-printing that is an option.